woocommerce/woocommerce-ios
# Adds "Draft" label below the image on the product editing screen

Fixes #4544

## Description

When editing a draft product we want to show a “Draft” label below the product image for consistency with the Android app:

![completed](https://user-images.githubusercontent.com/3812076/128674563-7b919c7f-d4e4-4602-8d74-2475e9db50aa.gif)

## Changes

| Draft products status  | Other product status |
|---------------------|-----------------|
| <img width="339" alt="Screenshot 2021-08-09 at 14 50 05" src="https://user-images.githubusercontent.com/3812076/128674613-322877c9-beb8-4565-8f31-039a5d794a9c.png"> | <img width="346" alt="Screenshot 2021-08-07 at 16 53 25" src="https://user-images.githubusercontent.com/3812076/128596361-258bafa2-3fe6-4222-82af-79b03f541567.png"> |

| Landscape View |
|---------------------|
| <img width="701" alt="Screenshot 2021-08-09 at 14 50 15" src="https://user-images.githubusercontent.com/3812076/128674740-47f29bf1-8c44-4e2b-b2d2-ce9474e48bbd.png"> |

| Example of LTR screen & Localized strings |
|---------------------|
| <img width="329" alt="Screenshot 2021-08-09 at 15 05 49" src="https://user-images.githubusercontent.com/3812076/128676201-02d32321-3203-4089-8d55-d499acc32036.png"> |



I've added a new `UILabel` to `ProductImagesHeaderTableViewCell.xib` via the Interface Builder . Then within the `ProductFormTableViewDataSource` I'm performing a check to see if the product status is a draft, so this label will only be displayed if the product status is set to `draft`, no label will be shown otherwise. 

I used existing styles that we use elsewhere in order to apply some sort of design, in this case using the same style as completed orders via `applyStyle(for: .completed)` seemed appropriate as fits the the style shown in the Android version of the app.

## Work in progress:
- [x] In order to fit as close as possible the Android one, I need to style it further by adding inner padding and changing the text colour.
- [x] Move the new label cell configuration outside the Data Source.
- [x] Fix UI constraints
- [x] Test for localized strings
- [x] Test for dark mode
- [x] Move the `.cell` appearance-only config to be called from `awakeFromNib()` instead, as these do not depend on data updates.

## Testing steps
- Create a new product, or edit any existing product.
- If the product is currently a `draft` , the `draft` label will appear below the image cell.
- In the top right corner, tap on "..." > Product Settings > Status
- Switching between different status will make the label appear only when `draft` is selected, and disappear on other statuses.

Update release notes:
